src/core/server/saved_objects/migrationsv2/model/model.ts

```diff
@@ -372,6 +373,31 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
     } else {
       return throwBadResponse(stateP, res.left);
     }
+  } else if (stateP.controlState === 'CALCULATE_EXCLUDE_FILTERS') {
+    const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
+
+    if (Either.isRight(res)) {
+      const unusedTypesQuery: estypes.QueryDslQueryContainer = {
+        bool: {
+          filter: [stateP.unusedTypesQuery, res.right.excludeFilter],
+        },
+      };
+
+      return {
+        ...stateP,
+        controlState: 'CREATE_REINDEX_TEMP',
+        unusedTypesQuery,
+        logs: [
+          ...stateP.logs,
+          ...res.right.errors.map((e) => ({
+            level: 'warning' as const,
+            message: `Ignoring excludeOnUpgrade hook that failed with error: "${e.toString()}"`,
```

Comment:
Can we attach info on what plugin or a query led to the problem? How are we supposed to identify that?